src/parcels/_core/converters.py

```diff
@@ -28,11 +29,16 @@ def _convert_to_flat_array(var: npt.ArrayLike) -> npt.NDArray:
 
 
 class UnitConverter:
-    """Interface class for spatial unit conversion during field sampling that performs no conversion."""
-
     source_unit: str | None = None
     target_unit: str | None = None
```

Comment:
Make into an abstract base class? This enforces that the subclasses have to implement the `to_target` and `to_source` methods to be valid

```diff
+from abc import ABC, abstractmethod

-class UnitConverter:
+class UnitConverter(ABC):
    """Interface class for spatial unit conversion during field sampling that performs no conversion."""

    source_unit: str | None = None
    target_unit: str | None = None

+	@abstractmethod
+     def to_target(self, value, z, y, x): ...

+     @abstractmethod
+     def to_source(self, value, z, y, x): ...
```